oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolver.java

```diff
@@ -36,7 +39,21 @@
  */
 public final class DefaultBearerTokenResolver implements BearerTokenResolver {
 
-	private static final Pattern authorizationPattern = Pattern.compile("^Bearer (?<token>[a-zA-Z0-9-._~+/]+)=*$");
+	private static final BearerTokenError BEARER_TOKEN_ERROR_MULTIPLE = new BearerTokenError(
+			BearerTokenErrorCodes.INVALID_REQUEST, HttpStatus.BAD_REQUEST,
+			"Found multiple bearer tokens in the request", "https://tools.ietf.org/html/rfc6750#section-3.1");
+
+	private static final BearerTokenError BEARER_TOKEN_ERROR_MALFORMED = new BearerTokenError(
+			BearerTokenErrorCodes.INVALID_TOKEN, HttpStatus.UNAUTHORIZED, "Bearer token is malformed",
+			"https://tools.ietf.org/html/rfc6750#section-3.1");
+
+	private static final String ACCESS_TOKEN_PARAM_NAME = "access_token";
+
+	private static final String TOKEN_CHARS = "(?<token>([a-zA-Z0-9-._~+/]+)=*$)";
```

Comment:
Earlier regex pattern was incorrect and was excluding `=` chars from extracted token hence updated to new regex, please review